go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go

```diff
@@ -280,3 +217,50 @@ func TestDMLOnNewTable(t *testing.T) {
 	defer utils.Exec(t, conn, `delete from t8`)
 	utils.AssertMatchesNoOrder(t, conn, `select id from new_table_tracked join t8`, `[[INT64(0)] [INT64(1)]]`)
 }
+
+// TestNewView validates that view tracking works as expected.
+func TestNewView(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	// insert some data
+	_ = utils.Exec(t, conn, "insert into t2 (id3, id4) values (1, 10), (2, 20), (3, 30)")
+	defer utils.Exec(t, conn, "delete from t2")
+
+	selQuery := "select sum(id4) from t2 where id4 > 10"
+
+	// create a view
+	_ = utils.Exec(t, conn, "create view test_view as "+selQuery)
+	time.Sleep(2 * time.Second)
+
+	// executing the query directly
+	qr := utils.Exec(t, conn, selQuery)
+	// selecting it through the view.
+	utils.AssertMatches(t, conn, "select * from test_view", fmt.Sprintf("%v", qr.Rows))
+}
+
+// TestViewAndTable validates that new column added in table is present in the view definition
+func TestViewAndTable(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	// insert some data
+	_ = utils.Exec(t, conn, "alter table t8 add column new_col varchar(50)")
+	time.Sleep(2 * time.Second)
+
+	_ = utils.Exec(t, conn, "insert into t8(id8, new_col) values (1, 'V')")
+	defer utils.Exec(t, conn, "delete from t8")
+
+	// create a view with t8, having the new column.
+	_ = utils.Exec(t, conn, "create view t8_view as select * from t8")
+	time.Sleep(2 * time.Second)
```

Comment:
Don't you want to create the view before adding the column? Otherwise isn't it just created when the table already has the new column?

Reply:
I can add that test, in that test, view should not contain the new column.